distributed/tests/test_as_completed.py

```diff
@@ -168,3 +170,107 @@ def test_async_for_py2_equivalent(c, s, a, b):
 
     with pytest.raises(StopAsyncIteration):
         yield seq.__anext__()
+
+
+@gen_cluster(client=True)
+def test_as_completed_error_async(c, s, a, b):
+        x = c.submit(throws, 1)
+        y = c.submit(sleep, 1)
+
+        ac = as_completed([x, y])
+        first = yield ac.__anext__()
+        second = yield ac.__anext__()
+        result = [first, second]
+
+        assert result == [x, y]
+        assert x.status == 'error'
+        assert y.status == 'finished'
```

Comment:
Two requests:

1.  Lets replace `sleep` by `inc` throughout in order to keep these tests fast if possible.  This will probably require making the tests insensitive to ordering issues as you did before
2.  Lets unindent the `gen_cluster` style tests one level

Otherwise things here look good to me.